src/allmydata/frontends/magic_folder.py

```diff
@@ -141,13 +142,20 @@ def load_magic_folders(node_directory):
                 interval = 60
             dir_fp = to_filepath(directory)
 
+            # make sure directory for magic folder exists
             if not dir_fp.exists():
-                raise Exception(
-                    "The '[magic_folder] local.directory' parameter is {} "
-                    "but there is no directory at that location.".format(
-                        quote_local_unicode_path(directory),
+                # there is something there, but it is not a directory,
+                # so we try to create one and cry if it fails
+                try:
+                    umask = config.get("magic_folder", "download.umask").decode('utf8')
+                    os.mkdir(quote_local_unicode_path(directory, 0777 & (~ umask)))
```

Comment:
New code should prefer `0o` syntax for octals, I suppose.  Update this to the new syntax, if you like.